src/server/clientiface.cpp

```diff
@@ -435,32 +417,38 @@ void RemoteClient::GotBlock(v3s16 p)
 
 void RemoteClient::SentBlock(v3s16 p)
 {
-	if (m_blocks_sending.find(p) == m_blocks_sending.end())
-		m_blocks_sending[p] = 0.0f;
-	else
+	if (!m_blocks_sending.insert(p).second)
 		infostream<<"RemoteClient::SentBlock(): Sent block"
 				" already in m_blocks_sending"<<std::endl;
 }
 
-void RemoteClient::SetBlockNotSent(v3s16 p)
+void RemoteClient::SetBlockNotSent(v3s16 p, bool low_priority)
 {
-	m_nothing_to_send_pause_timer = 0;
-
 	// remove the block from sending and sent sets,
-	// and mark as modified if found
-	if (m_blocks_sending.erase(p) + m_blocks_sent.erase(p) > 0)
-		m_blocks_modified.insert(p);
+	// and reset the scan loop if found
+	if (m_blocks_sending.erase(p) + m_blocks_sent.erase(p) > 0) {
+		m_nothing_to_send_pause_timer = 0;
+		// If this is a low_pririty event, do not reset m_nearest_unsent_d.
+		// Instead, the send loop will get to the block in the next full loop iteration.
+		if (!low_priority) {
+			// Note that we do NOT use the euclidean distance here.
+			// getNextBlocks builds successive cube-surfaces in the send loop.
+			// This resets the distance to the maximum cube size that
+			// still guarantees that this block will be scanned again right away.
+			//
+			// Using m_last_center is OK, as a change in center
+			// will reset m_nearest_unsent_d to 0 anyway (see getNextBlocks).
+			p -= m_last_center;
+			s16 this_d = (s16)std::max(std::max(std::abs(p.X), std::abs(p.Y)), std::abs(p.Z));
```

Comment:
```suggestion
			s16 this_d = std::max({std::abs(p.X), std::abs(p.Y), std::abs(p.Z)});
```

Reply:
I suppose this is slight more expensive as it has construct an array first... But I think that's fine, and it's more readable.